deckhouse-controller/pkg/controller/controller.go

```diff
@@ -201,10 +265,118 @@ func (dml *DeckhouseController) runEventLoop(ec chan events.ModuleEvent) {
 				log.Errorf("Error occurred during the module %q turning off: %s", mod.GetBasicModule().GetName(), err)
 				continue
 			}
+
+		case events.ModuleStateChanged:
+			err := dml.updateModuleStatus(event.ModuleName)
+			if err != nil {
+				log.Errorf("Error occurred during the module %q status update: %s", event.ModuleName, err)
+				continue
+			}
 		}
 	}
 }
 
+func (dml *DeckhouseController) updateModuleConfigStatus(configName string) error {
+	return retry.OnError(retry.DefaultRetry, errors.IsServiceUnavailable, func() error {
+		metricGroup := fmt.Sprintf("%s_%s", "obsoleteVersion", configName)
+		dml.metricStorage.GroupedVault.ExpireGroupMetrics(metricGroup)
+		moduleConfig, err := dml.kubeClient.DeckhouseV1alpha1().ModuleConfigs().Get(dml.ctx, configName, v1.GetOptions{})
+
+		// if module config found
+		if err == nil {
+			newModuleConfigStatus := d8config.Service().StatusReporter().ForConfig(moduleConfig)
+			if (moduleConfig.Status.Message != newModuleConfigStatus.Message) || (moduleConfig.Status.Version != newModuleConfigStatus.Version) {
+				patch, err := json.Marshal([]moduleConfigPatch{{
```

Comment:
```suggestion
				patch, _ := json.Marshal([]moduleConfigPatch{{
```

we can omit an error here

Reply:
remove the check but I'm not sure why :) like we are 100% sure that it will marshal successfully